hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/AbstractReplicatedRecordStore.java

```diff
@@ -75,7 +75,7 @@ public Object removeWithVersion(Object key, long version) {
     @SuppressWarnings("unchecked")
     private Object remove(InternalReplicatedMapStorage<K, V> storage, Object key) {
         isNotNull(key, "key");
-        long time = Clock.currentTimeMillis();
+        long time = System.nanoTime();
```

Comment:
Changes in this class seem to be unrelated. #16781 is meant to address the problem across the whole code base.